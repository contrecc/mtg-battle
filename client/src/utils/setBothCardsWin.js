export default function setBothCardsWin() {
  return function(state, props) {
    return {
      bothStrong: true,
      winStreak: state.winStreak + 1,
      winningCard: '',
      winner: false,
      roundCompleted: true
    };
  };
}
