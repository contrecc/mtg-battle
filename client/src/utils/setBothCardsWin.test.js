import setBothCardsWin from './setBothCardsWin';

describe('setBothCardsWin', () => {
  it('should return a function', () => {
    expect(setBothCardsWin()).toMatchSnapshot();
  });

  it('should return a function that then updates the state', () => {
    const returnedFunction = setBothCardsWin();
    expect(returnedFunction({}, {})).toMatchSnapshot();
  });
});
