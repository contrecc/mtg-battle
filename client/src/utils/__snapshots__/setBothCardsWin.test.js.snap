// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setBothCardsWin should return a function 1`] = `[Function]`;

exports[`setBothCardsWin should return a function that then updates the state 1`] = `
Object {
  "bothStrong": true,
  "roundCompleted": true,
  "winStreak": NaN,
  "winner": false,
  "winningCard": "",
}
`;
